src/common/lib/client/restchannelmixin.ts

```diff
@@ -36,13 +37,11 @@ export class RestChannelMixin {
       headers,
       unpacked,
     ) {
-      return await messageFromResponseBody(
-        body as Message[],
-        options,
-        channel.logger,
-        client._MsgPack,
-        unpacked ? undefined : format,
-      );
+      const decoded: WireProtocolMessage[] = unpacked
+        ? (body as WireProtocolMessage[])
+        : Utils.decodeBody(body, client._MsgPack, format);
+
+      return fromEncodedArray(channel.logger, Platform.Crypto, decoded, options);
```

Comment:
`Platform.Crypto` is not set in the modular variant of the SDK, so I don't think this will work. It sounds like you're saying that you observed an issue with history messages not getting decrypted; it might be worth raising a separate issue and adding some tests to demonstrate the problem?

Reply:
shouldn't have implied there was an issue in the current ver, sorry -- this was a yakshave from removing fromResponseBody, which I wanted to do because relying on whether the format parameter was set to determine whether its argument was encoded or not struck me as very messy, and anyway the typings weren't at all right for that. I thought I could just substitute fromEncodedArray, which basically does the same thing. I'm not really familar with the modular variant implementation; having looked into it briefly it seems like I could just use client._Crypto, but might be neater to go back to having a separate form of the parser for when you have a channel object already so don't need to renormalize channel options. I'll do that.

Reply:
done

and added a rest encrypted history test

Reply:
Sorry, I’m still trying to unpick what's going on in c406bef8bc3b27ca5d4f92d540066384c3ae5751; specifically:

- what necessitated the added fetches of `Platform.Crypto`; as I mentioned above, in the modular variant of the library this is unlikely to have whatever effect that it is you're trying to achieve, but in order to tell you what to replace it with I need to understand better why they were added
- the part of the commit message that says "crypto not being passed-in to history calls, so not getting decrypted", which I’m reading as saying that there is some sort of an issue with history decryption. In order to satisfy us that there is _not_ an issue with history decryption, and to allow us to limit the scope of this commit to just being a refactor instead of refactor + maybe a fix to history, I’ve split out your added history test into a separate PR and added a similar test for the modular variant of the library (https://github.com/ably/ably-js/pull/1933)

Reply:
> in order to tell you what to replace it with I need to understand better why they were added

The Platform.Crypto additions divided into three categories:
1. The ones in restpresence/restpresencemixin/restchannel/restchannelmixin, replacing the previous `fromResponseBody`. As described [above](https://github.com/ably/ably-js/pull/1923#discussion_r1861380400) in response to your last comment I removed those and replaced them with an underscore-prefixed form of `fromEncodedArray` for use when you already have a channel object.
2. the addition to defaultpresencemessage.ts. This states that it's used in the non-tree-shakable version of the library -- and also Platform.Crypto was already used in exactly the same way in `defaultmessage.ts`, for which `defaultpresencemessage.ts` is just the presence version. It just looked the tree-shakability work assumed that presence events could not have encrypted data, but of course this is false.
3. The addition to `web/modular/presencemessage.ts`. This was similar to the previous option -- it looks like the lib was assuming presence messages could not be encrypted -- but I made the mistake of copying the same fix I'd used in defaultpresencemessage here, when I should have looked at what was done in `web/module/message.ts`. I've now done that, and copied the approach there: 24f6cea4b361c29d3f448065eb1816182ffe1ca3

I've also updated the commit message to avoid misleading mentions of history, as mentioned history was not the problem that was in main (though I guess not decrypting presence messages is, at least in some versions).

Reply:
> Platform.Crypto was already used in exactly the same way in defaultmessage.ts

Yep, it's fine for these `Default*` classes to access `Platform.Crypto`.

> it looks like the lib was assuming presence messages could not be encrypted

Yep, that was an oversight. Might be a good idea to add tests for your new `decodeEncryptedPresenceMessage*` functions? (you can copy the tests of the `decodeEncryptedMessage*` functions in `modular.test.js`, although I don't know if we have test fixtures for encrypted presence messages?)

Could we also please do a couple of things to make d863e91be85384a010b9ad915803d8009369c6d8 understandable for future ourselves, namely:

- remove the passing of `Platform.Crypto` in `src/platform/web/modular/realtimepresence.ts` (I know you undo this in 29d1ed9b7e51f5febf15d95c02520f36dfdcaf6c, but it's a bit confusing when looking at d863e91be85384a010b9ad915803d8009369c6d8 in isolation)
- drop the encrypted history test and the shared helper change that supports it, because it's not really clear how they're related to this commit now? we can get this test from #1933 instead.

Reply:
Done all.

(no particular need for encrypted presence message specific fixtures, looks like the test just works from plain objects with `data` fields and compares the data field of the result, so the messages and presence messages versions are doing exactly the same thing)